### PR TITLE
Fix adaptive repartition join leak

### LINKED ISSUE #3459: Query cancellation with re-partition joins via adaptive executor leak intermediate files
```
while True:
   - Start a long running re-partition join via adaptive executor. 
   - Cancel the query after 5 seconds
   - Check the data size under `base/pgsql_job_cache` with `du -sh  on any of the worker nodes.` You'll see that the size keeps increasing.
```

### PR DESCRIPTION
**Pull request by serprex**
Fixes #3459

First commit: typo noticed while reviewing #3169 
Second commit: connectionId in MultiConnection added by #3169 seems to be only used for one pre-existing log message. Debug code? Removed
Third commit: fix jobIdList never being set for DistributedExecution

### CODE REVIEW

**Comment by pykello** [comment]
Is it possible to add some failure tests for this?

File: src/backend/distributed/connection/remote_commands.c [hunk 1]
```diff
@@ -353,9 +353,8 @@ LogRemoteCommand(MultiConnection *connection, const char *command)
 	}
 
 	ereport(NOTICE, (errmsg("issuing %s", ApplyLogRedaction(command)),
-					 errdetail("on server %s@%s:%d connectionId: %ld", connection->user,
-							   connection->hostname,
-							   connection->port, connection->connectionId)));
```
Comment (marcocitus) [inline src/backend/distributed/connection/remote_commands.c]: We've been using connectionId heavily to debug/understand connection management issues.

Comment (serprex) [reply src/backend/distributed/connection/remote_commands.c]: Good to know, that's a discussion for another time then. Dropping that commit from this PR

**Comment by SaitTalhaNisanci** [comment]
I think this fixes the case for pg_catch but if something doesnt enter pg_catch and receives a cancel signal. We still do not clean up the intermediate files, am I wrong?

**Comment by marcocitus** [comment]
> I think this fixes the case for pg_catch but if something doesnt enter pg_catch and receives a cancel signal. We still do not clean up the intermediate files, am I wrong?

In that case we go into the regular call to DoRepartitionCleanup in AdaptiveExecutor.

**Comment by marcocitus** [comment]
This doesn't cover all cases. The cancellation can happen while executing the dependencies, rather than at the top level. At the bottom of the tree, the jobIdList is empty. I wonder whether we should set some global variable and clean up in the abort handler.

**Comment by serprex** [comment]
@marcocitus I think the best way forward then is to merge this PR & I'll continue looking into a more complete fix

**Comment by marcocitus** [comment]
> @marcocitus I think the best way forward then is to merge this PR & I'll continue looking into a more complete fix

If we move the clean-up to the abort handler then we will probably not need to DoRepartitionCleanup in the PG_CATCH at all.

**Comment by marcocitus** [comment]
I get the following compile error:
```
executor/distributed_intermediate_results.c: In function ‘ExecuteSelectTasksIntoTupleStore’:
executor/distributed_intermediate_results.c:415:2: error: too few arguments to function ‘ExecuteTaskListExtended’
  ExecuteTaskListExtended(ROW_MODIFY_READONLY, taskList, resultDescriptor,
  ^~~~~~~~~~~~~~~~~~~~~~~
In file included from executor/distributed_intermediate_results.c:24:
/home/marco/citus/citus-fixleak2/src/include/distributed/multi_executor.h:76:15: note: declared here
 extern uint64 ExecuteTaskListExtended(RowModifyLevel modLevel, List *taskList,
```

**Comment by marcocitus** [comment]
Solution looks good (enough), though could really use a failure test.

**Pull request merged**